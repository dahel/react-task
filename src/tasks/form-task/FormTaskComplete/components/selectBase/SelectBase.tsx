import { SyntheticEvent } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  min-width: 200px;
`;

const Label = styled.label`
  padding: 0 0 5px 5px;
`;

const Select = styled.select`
  padding: 5px;
  border: solid 1px #bdbdbd;
  border-radius: 4px;
`;
interface IProps {
  value: string;
  label: string;
  name: string;
  options: ISelectBaseOptions[];
  onChange?: (event: SyntheticEvent) => void;
}

export interface ISelectBaseOptions {
  value: string;
  label: string;
}

const SelectBase = ({ value, label, name, onChange = () => {}, options = [] }: IProps) => {
  return (
    <Wrapper>
      <Label htmlFor={name}>{label}</Label>
      <Select name={name} id={name} value={value} onChange={onChange}>
        {options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </Select>
    </Wrapper>
  );
};

export default SelectBase;
