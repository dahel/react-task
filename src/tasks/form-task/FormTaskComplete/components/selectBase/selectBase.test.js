import renderer from 'react-test-renderer';
import SelectBase from './SelectBase';

const props = {
  name: 'name',
  label: 'label',
  value: 'value',
  options: [
    {
      value: 'option1',
      label: 'label1',
    },
    {
      value: 'option2',
      label: 'label2',
    },
  ],
};

test('renders properly for only-required props passed', () => {
  const component = renderer.create(<SelectBase {...props} />);

  expect(component.toJSON()).toMatchSnapshot();
});

test(`calls 'onChange' callback passed as parameter`, () => {
  const onChange = jest.fn();
  const root = renderer.create(<SelectBase {...props} onChange={onChange} />).root;
  const select = root.findByType('select');

  select.props.onChange({ name: 'change' });

  expect(onChange).toHaveBeenCalledWith({ name: 'change' });
});
