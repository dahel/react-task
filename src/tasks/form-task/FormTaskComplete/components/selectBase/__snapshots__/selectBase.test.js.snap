// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly for only-required props passed 1`] = `
<div
  className="sc-bdvvtL cTwBbG"
>
  <label
    className="sc-gsDKAQ cSZgOM"
    htmlFor="name"
  >
    label
  </label>
  <select
    className="sc-dkPtRN gjymBZ"
    id="name"
    name="name"
    onChange={[Function]}
    value="value"
  >
    <option
      value="option1"
    >
      label1
    </option>
    <option
      value="option2"
    >
      label2
    </option>
  </select>
</div>
`;
